### PR TITLE
Introduce isort.

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**
Add `isort` to `requirements-dev.txt` and run before `black` upon `make format` to sort Python imports.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Nice!

**Review by jay-selby — APPROVED** [review]
Nice!  I'm not sure about how it decides to add whitespace, but at least it is consistent

**Comment by AmintorDusko** [comment]
> Nice! I'm not sure about how it decides to add whitespace, but at least it is consistent

What more can we ask? :joy:

File: Makefile [hunk 1]
```diff
@@ -111,6 +111,7 @@ format-cpp:
 	./bin/format $(CHECK) ./pennylane_lightning
 
 format-python:
+	isort ./pennylane_lightning/ ./mpitests ./tests $(CHECK)
```
Comment (math411) [inline Makefile:114]: Using `--profile black` could help reduce redundant work done with formatting: https://pycqa.github.io/isort/docs/configuration/black_compatibility.html

Comment (vincentmr) [reply Makefile]: Thanks @math411 , I didn't know about this option. Sure is useful for `black` users.

**Pull request merged**